spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizer.java

```diff
@@ -236,6 +242,14 @@ public Integer doInSession(Session<F> session) throws IOException {
 					F[] files = session.list(remoteDirectory);
 					if (!ObjectUtils.isEmpty(files)) {
 						List<F> filteredFiles = filterFiles(files);
+						if (maxFetchSize >= 0 && filteredFiles.size() > maxFetchSize) {
+							rollbackFromFileToListEnd(filteredFiles, filteredFiles.get(maxFetchSize));
+							List<F> newList = new ArrayList<>(maxFetchSize);
+							for (int i = 0; i < maxFetchSize; i++) {
+								newList.add(filteredFiles.get(i));
+							}
+							filteredFiles = newList;
```

Comment:
OK. I think we should NOTE this `0` case in the Docs. Because that can be surprise for end-user do not receive anything all the time with `0` if he/she doesn't  change it to the proper value eventually 😄